File: 1582. Special Positions in a Binary Matrix/solution.java
class Solution {
  public int numSpecial(int[][] mat) {
    int[] rowAmount = new int[mat[0].length];
    Arrays.fill(rowAmount, 0);
    int[] columnAmount = new int[mat.length];
    Arrays.fill(columnAmount, 0);

    for (int row = 0; row < mat.length; row++){
      for (int column = 0; column < mat[row].length; column++){
        int value = mat[row][column];
        rowAmount[column] += value;
        columnAmount[row] += value;
      }
    }

    int result = 0;
    for (int row = 0; row < mat.length; row++){
      for (int column = 0; column < mat[row].length; column++){
        int value = mat[row][column];
        if (value == 1 && rowAmount[column] == 1 && columnAmount[row] == 1){
          result += 1;
        }
      }
    }

    return result;
  }
}

File: 1582. Special Positions in a Binary Matrix/solution.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
 var numSpecial = function(mat) {
  let rowAmount = new Array(mat[0].length).fill(0);
  let columnAmount = new Array(mat.length).fill(0);

  for (let row = 0; row < mat.length; row++){
    for (let column = 0; column < mat[row].length; column++){
      let value = mat[row][column];
      rowAmount[column] += value;
      columnAmount[row] += value;
    }
  }

  let result = 0;
  for (let row = 0; row < mat.length; row++){
    for (let column = 0; column < mat[row].length; column++){
      let value = mat[row][column];
      if (value === 1 && rowAmount[column] === 1 && columnAmount[row] === 1){
        result += 1;
      }
    }
  }

  return result;
};